ykrt/src/mt.rs

```diff
@@ -366,7 +366,13 @@ impl MT {
                 trace_iter.2,
             ) {
                 Ok(ct) => {
+                    let addr = ct.entry();
+                    // The order of the following two calls is important to avoid a race condition.
+                    // If we were to patch the parent trace first, there is a small window where
+                    // another thread takes the patched jump and deopts before we had a chance to
+                    // call `set_ctr` which sets information required by deopt.
```

Comment:
I think there is still a race here:

1. Thread A creates a side trace for guard G, calls `set_ctr` and is then interrupted.
2. Thread B also creates a side trace for G, calls `set_ctr` and `patch_guard`.
3. Thread A wakes up and calls `patch_guard.

Since `set_ctr` grabs a lock on G, why not use that to also call `patch_guard`? AFAICS we never call `set_ctr` without also calling `patch_guard`.

Reply:
Is this what you had in mind? [03d3eb6](https://github.com/ykjit/yk/pull/1591/commits/03d3eb6250734d9a1b84a7f7af2e6f1b14690bbe)

Reply:
Yes. We should update the doc string too, but otherwise I think this fixes the probable race.

Reply:
Like so [7341847](https://github.com/ykjit/yk/pull/1591/commits/734184788a4609dbda62713fcada126471c0ad23)?

Reply:
That's very weird formatting which might cause `cargo doc` or `clippy` to bork (they're quite fussy). I'd suggest either making those part of the paragraph *or* using `*` bullet points. Please squash and make whichever update you see fit.

Reply:
Neither cargo doc nor clippy complained. But as my lord wishes: [c05f68e](https://github.com/ykjit/yk/pull/1591/commits/c05f68e9d5d085cb02010584f96aed104e7aef2b) [0f74ef3](https://github.com/ykjit/yk/pull/1591/commits/0f74ef3b5fc092ba90e6bf24ec9351eda5eb29cb)